hooks/shapediver/viewer/interaction/useSelectManager.ts

```diff
@@ -0,0 +1,102 @@
+import { ISelectionParameterSettings, MaterialStandardData } from "@shapediver/viewer";
+import { SelectManager, MultiSelectManager, InteractionEngine } from "@shapediver/viewer.features.interaction";
+import { useEffect } from "react";
+import { useInteractionEngine } from "./useInteractionEngine";
+
+// #region Functions (1)
+
+// create an object to store the select managers or the multi select managers for the viewports
+const selectManagers: { [key: string]: {
+	selectManager: SelectManager | MultiSelectManager,
+	selectMultiple: boolean,
+	token: string
+} } = {};
+
+/**
+ * Clean up the select manager for the given viewportId.
+ * We also deselect all selected nodes.
+ *
+ * @param viewportId - The ID of the viewport.
+ * @param interactionEngine - The interaction engine instance.
+ */
+const cleanUpSelectManager = (viewportId: string, interactionEngine?: InteractionEngine) => {
+	if (selectManagers[viewportId]) {
+		if (selectManagers[viewportId].selectManager instanceof SelectManager) {
+			selectManagers[viewportId].selectManager.deselect();
+		} else if (selectManagers[viewportId].selectManager instanceof MultiSelectManager) {
+			selectManagers[viewportId].selectManager.deselectAll();
+		}
+		if(interactionEngine)
+			interactionEngine.removeInteractionManager(selectManagers[viewportId].token);
+		delete selectManagers[viewportId];
+	}
+};
+
+/**
+ * Hook allowing to create a select manager for a viewport.
+ * 
+ * @param viewportId The ID of the viewport.
+ * @param settings The settings for the select manager. If the settings are not provided, the select manager will not be created.
+ */
+export function useSelectManager(viewportId: string, settings?: ISelectionParameterSettings): {
+	/**
+	 * The select manager that was created for the viewport.
+	 * Depending on the settings, this can be a select manager or a multi select manager.
+	 */
+	selectManager?: SelectManager | MultiSelectManager
+} {
+	// call the interaction engine hook
+	const { interactionEngine } = useInteractionEngine(viewportId);
+
+	// use an effect to create the select manager
+	useEffect(() => {
+		if (settings) {
+			// whenever this output node changes, we want to create the interaction engine
+			const selectMultiple = (settings.minimumSelection !== undefined && settings.maximumSelection !== undefined) &&
+				settings.minimumSelection < settings.maximumSelection && settings.maximumSelection > 1;
+
+			// check if a select manager already exists for the viewport, but with different settings
+			// in this case we need to remove the old select manager and create a new one
+			if (selectManagers[viewportId] && selectManagers[viewportId].selectMultiple !== selectMultiple) {
+				cleanUpSelectManager(viewportId, interactionEngine);
+			}
```

Comment:
@MajorMeerkatThe3rd there might be situations were there are two selection inputs, both activated, one of the for one object only, the other one for multiple objects. This would cause a problem. For now we can issue a warning to the console, but can we maybe use `MultiSelectManager` only?

Reply:
@snabela For now that would be a general problem. As there is no assignment for a selection engine to the node that it can intersect it means that any node with an interaction data can be selected by any selection manager. This has to be implemented on the viewer side first.

Reply:
@MajorMeerkatThe3rd got it. We can test and fix this as an extension. Probably we will use some global state to check whether another selection is already ongoing, and block activating a second one.

Reply:
https://shapediver.atlassian.net/browse/SS-7920